pkgs/top-level/aliases.nix

```diff
@@ -1054,6 +1054,7 @@ mapAliases ({
 
   ### N ###
 
+  nbqa = lib.warn "nbqa has been moved to python3Packages.nbqa" python3Packages.nbqa; # Added 2024-09-20
```

Comment:
If we want to follow, how things are done with `isort`, `black` and other similar packages, this should probably be
```nix
nbqa = with python3Packages; toPythonApplication nbqa;
```

Reply:
I'll still keep the `lib.warn`, but I'll consider adding `toPythonApplication`.

Reply:
Having it available both as `pkgs.nbqa` and `pkgs.python3Packages.nbqa` has value - some users just want to use the CLI, and others actually want to interact with it via Python.

Reply:
Are you sure? The `X = with python3Packages; toPythonApplication X;` pattern is used 221 times in `pkgs/top-level/all-packages.nix` without any warnings. To me, a warning indicates that this is a "deprecated" alias and the user should change their usage to the new location.

In this case, `pkgs.nbqa` isn't an alias for `pkgs.python3Packages.nbqa`. One is a standalone application, the other is a python module/package (intended to be installed in a python environment).

Reply:
> Having it available both as pkgs.nbqa and pkgs.python3Packages.nbqa has value - some users just want to use the CLI, and others actually want to interact with it via Python.

You can use the CLI of packages built with `buildPythonPackage`. It's just that the application won't be in the top-level namespace.

```
nixpkgs on  move-nbqa-to-python3packages [$] took 17s 
❯ nix build .#python3Packages.nbqa

nixpkgs on  move-nbqa-to-python3packages [$] took 5s 
❯ result/bin/nbqa 
usage: nbqa <code quality tool> <notebook or directory> <nbqa options> <code quality tool arguments>

Please specify:
- 1) a code quality tool (e.g. `black`, `pyupgrade`, `flake`, ...)
- 2) some notebooks (or, if supported by the tool, directories)
- 3) (optional) flags for nbqa (e.g. `--nbqa-diff`, `--nbqa-shell`)
- 4) (optional) flags for code quality tool (e.g. `--line-length` for `black`)

Examples:
    nbqa flake8 notebook.ipynb
    nbqa black notebook.ipynb --line-length=96
    nbqa pyupgrade notebook_1.ipynb notebook_2.ipynb

See https://nbqa.readthedocs.io/en/latest/index.html for more details on how to run `nbqa`.
nbqa: error: the following arguments are required: command, root_dirs
```

Reply:
I guess this boils down to whether there's use cases for nbqa as a library.

Reply:
The author of the linked issue appears to have had a need to use it as a library, although it might have been an XY problem.

Reply:
The author of the linked issue appears to have had a need to use it as a library, although it might have been an XY problem.

Reply:
> others actually want to interact with it via Python.

Can you show me proof that nbqa has a public Python API?